### PR TITLE
fix: clmul aarch64

### PR DESCRIPTION
**Pull request by sinui0**
This PR reorganizes some things and brings the `clmul` crate more inline with the original `polyval` crate.

# Changes
- Removes `armv8` and `force-soft` features in favor of using `RUSTFLAGS`
- Reorganizes modules to match `polyval` layout

### CODE REVIEW

File: clmul/src/backend/autodetect.rs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+            }
+        }
+    }
+}
+
+impl Clmul {
+    pub fn new(h: &[u8; 16]) -> Self {
+        let (token, has_intrinsics) = mul_intrinsics::init_get();
+
+        let inner = if cfg!(clmul_force_soft) || !has_intrinsics {
```
Comment (themighty1) [inline clmul/src/backend/autodetect.rs:56]: if we don't have `intrinsics` and we didn't pass `clmul_force_soft`, this code will select `intrinsics` in the else{} branch below. But we need `soft` instead.

File: clmul/src/backend/autodetect.rs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            )
+        }
+    }
+
+    /// Performs carryless multiplication. Same as clmul() but reusing the
+    /// operands to return the result. This gives a ~6x speed up compared
+    /// to clmul() where we create new objects containing the result.
+    /// The high bits will be placed in `self`, the low bits - in `x`.
+    #[inline]
+    pub fn clmul_reuse(&mut self, x: &mut Self) {
```
Comment (themighty1) [inline clmul/src/backend/autodetect.rs]: Im in favor of getting rid of this fn since the speedup doesnt apply anymore.

File: clmul/Cargo.toml [hunk 3]
```diff
@@ -17,10 +17,6 @@ rand.workspace = true
 rand_core.workspace = true
 criterion.workspace = true
 
-[features]
```
Comment (themighty1) [inline clmul/Cargo.toml:20]: There are some refs to `armv8` and `force-soft` dangling in the code and comments. 
Can we also document the new `clmul_armv8` and `clmul_force_soft` flags? We can document them here as a comment.

**Review by themighty1 — CHANGES_REQUESTED** [review]
lgtm, a fix is needed

File: clmul/src/backend/autodetect.rs (hunk 1 above)
Comment (sinui0) [reply clmul/src/backend/autodetect.rs:56]: Are you sure? This evaluates true if
 - force soft is enabled, or
 - instrinsics aren't available

File: clmul/src/backend/autodetect.rs (hunk 2 above)
Comment (sinui0) [reply clmul/src/backend/autodetect.rs]: I agree

File: clmul/Cargo.toml (hunk 3 above)
Comment (sinui0) [reply clmul/Cargo.toml:20]: I documented them in the crate root docs

File: clmul/src/backend/autodetect.rs (hunk 1 above)
Comment (themighty1) [reply clmul/src/backend/autodetect.rs:56]: you are right, sry

**Review by themighty1 — APPROVED** [review]
lgtm

**Pull request merged**